### PR TITLE
Micah jank

### PR DESCRIPTION
**Pull request by MicahJank**

### CODE REVIEW

File: todo-app/src/reducers/reducer.js [hunk 1]
```diff
@@ -15,16 +15,23 @@ export const reducer = (state, action) => {
...
-                return state.map(todo => {
-                    if(todo.id === action.id) {
-                        return {...todo, complete: !todo.complete}
-                    } else {
-                        return todo;
-                    }
-                });
-            case 'CLEAR_COMPLETED':
-                return state.filter(todo => todo.complete === false);
+        case 'TOGGLE_COMPLETE':
```
Comment (MicahJank) [inline todo-app/src/reducers/reducer.js:72]: So i am trying to get it so my menu tabs in the Navigation will filter the different Todos by ALL, ACTIVE or COMPLETED. But the problem is that when the menu buttons onClick dispatches, the state gets overwritten i guess? Basically if i switch the menu to filter only by the 'completed' todos, then when i switch back to 'all' todos...it still ONLY SHOWS the completed todos...

Comment (MicahJank) [reply todo-app/src/reducers/reducer.js:72]: @jacobAngulo

Comment (jacobAngulo) [reply todo-app/src/reducers/reducer.js:72]: Hey @MicahJank, we went over this in standup, let me know if this is a new issue and I'll take a closer look.

File: todo-app/src/App.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import React from 'react';
+
+import Form from './components/Form.js';
+import TodoList from './components/TodoList.js';
+import HeaderTitle from './components/HeaderTitle.js';
+import Navigation from './components/Navigation.js';
+
+import { Container, Header } from 'semantic-ui-react';
```
Comment (jacobAngulo) [inline todo-app/src/App.js:8]: doesn't look like Header is getting used anywhere in this file

File: todo-app/src/components/Navigation.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+const Navigation = () => {
+
+    const [activeItem, setActiveItem] = useState('ALL');
+
+    const { dispatch } = useContext(TodoContext);
+
+    const handleClick = (e) => {
+        setActiveItem(e.target.getAttribute('value'));
+        dispatch({
+            type: `FILTER_${e.target.getAttribute('value')}`
```
Comment (jacobAngulo) [inline todo-app/src/components/Navigation.js:39]: this is super cool, but also, super broken

File: todo-app/src/reducers/reducer.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+                }
+            });
+        case 'CLEAR_COMPLETED':
+            return state.filter(todo => todo.complete === false);
+        case 'FILTER_COMPLETED':
+            return state.filter(todo => todo.complete === true);
+        case 'FILTER_ALL':
+            return state;
+        case 'FILTER_ACTIVE':
+            return state.filter(todo => todo.complete === false);
```
Comment (jacobAngulo) [inline todo-app/src/reducers/reducer.js]: You can see here that these reducers are  changing the state that the useReducer hook is using in the TodoProvider

**Review by jacobAngulo — APPROVED** [review]
Nice work Micah, let me know where I can elaborate / focus.

**Pull request merged**